x/deployment/keeper/keeper.go

```diff
@@ -311,9 +311,15 @@ func (k Keeper) OnPauseGroup(ctx sdk.Context, group types.Group) error {
 	store.Delete(key)
 
 	group.State = types.GroupPaused
+
+	key, err := GroupKey(GroupStateToPrefix(group.State), group.ID)
+	if err != nil {
+		return fmt.Errorf("%s: failed to encode group key", err)
+	}
+
```

Comment:
Looking at `OnCloseGroup`, it seems like the new key with the updated prefix should have been used here or else it gets saved to the wrong part of the KVStore. 

It does look like findGroup was checking all possible variants of a key when looking for a group, so it probably could have been found anyways, but this should still be a more accurate state/key for the Group